### PR TITLE
[DEV-465] Bug: a user can claim the same badge twice

### PR DESCRIPTION
**Pull request by grp06**
## Describe your changes
Bug: in the current app we use `active`, `passive`, and `uri` to generate a `tokenId`. The problem with this is that if the issuer of a badge changes but the recipient stays the same, then the recipient could claim two of the same badge. 

We fix this by changing how we calculate the hash of a `tokenId`. Now, we create the hash in the mint function based only on the `to` address and the `uri`

Now that we have this concept of a `voucherHashId` we need to create a mapping of `voucherHashId`s to `tokenId`s ... called `voucherHashId`. We need this in order to unequip and burn badges

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly. (if applicable)
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by grp06** [comment]
Still to-do: 
- fix snapshot tests
- add fuzz tests for checking hashes

File: test/badges.test.ts [hunk 1]
```diff
@@ -344,6 +344,57 @@ describe('Badges', async function () {
     mintBadge()
   })
 
+  it.only('should fail to mint a second badge after the claimant gets permission from another Raft owner under the same spec', async function () {
```
Comment (rahulrumalla) [inline test/badges.test.ts:347]: renam
```suggestion
  it.only('should fail when trying to claim using a voucher from another issuer for the same spec', async function () {
```

File: src/Badges.sol [hunk 2]
```diff
@@ -188,6 +189,10 @@ contract Badges is
     return _hashTypedDataV4(structHash);
   }
 
+  function getBadgeIdHash(address _to, string memory _uri) public view virtual returns (bytes32) {
+    return keccak256(abi.encode(_to, bytes(_uri)));
```
Comment (rahulrumalla) [inline src/Badges.sol:193]: remote casting to bytes

**Review by rahulrumalla — APPROVED** [review]

**Comment by rahulrumalla** [comment]
Hey. Let's merge this but not upgrade the contracts just yet. Let's keep track of this changelog and upgrade to 3.0.0 with all the other changes.

**Review by rahulrumalla — APPROVED** [review]

**Pull request merged**